README.md

```diff
@@ -14,83 +16,179 @@ yarn add --dev eslint-config-yscope
-
-#### Example Usage
-
-If your project uses Jest, you can add an override for your test files as follows:
+Some configurations are provided as standalone configuration objects, which can be directly
+referenced once imported:
+- `CommonConfig`: A general ESLint configuration suitable for a wide range of JavaScript projects.
+- `JestConfig`: A Jest-specific ESLint configuration intended to override other configurations for 
+Jest test files.
+- `MeteorConfig`: A specialized ESLint configuration for Meteor projects.
+
+The other configurations are provided as a group of configuration objects stored in an array. To use
+these configurations, you need to utilize the "spread" (`...`) syntax to unpack the configuration
+objects into your own configuration array:
+- `StylisticConfigArray`: A configuration for stylistic formatting. It is typically combined with 
+- `CommonConfig`, as shown in the [ESLint config examples](#eslint-config-examples).
+- `ReactConfigArray`: A specialized ESLint configuration tailored for React applications.
+- `TsConfigArray`: A configuration designed specifically for TypeScript projects. A helper 
+`createTsConfigOverride` is also provided. Refer to the
+[ESLint config examples](#eslint-config-examples) for details on how to set it up.
+
+
+### ESLint config examples
+
+1. For use with a CommonJS Node.js project:
+    ```javascript
+    import globals from "globals";
+    
+    import CommonConfig from "./CommonConfig.mjs";
+    import StylisticConfigArray from "./StylisticConfigArray.mjs";
+    
+    
+    const EslintConfig = [
+        CommonConfig,
+        ...StylisticConfigArray,
+        {
+            languageOptions: {
+                globals: {
+                    ...globals.node,
+                },
+            },
+        },
+        {
+            rules: {
+                // ... Your own overrides
+            },
+        },
+    ];
+    
+    
+    export default EslintConfig;
+    ```
+
+2. For use with a JavaScript React.js project:
+    ```javascript
+    import CommonConfig from "eslint-config-yscope/CommonConfig.mjs";
+    import ReactConfigArray from "eslint-config-yscope/ReactConfigArray.mjs";
+    import StylisticConfigArray from "eslint-config-yscope/StylisticConfigArray.mjs";
+    
+    
+    const EslintConfig = [
+        CommonConfig,
+        ...StylisticConfigArray,
+        ...ReactConfigArray,
+    ];
+    
+    
+    export default EslintConfig;
+    ```
+
+3. For use with a TypeScript React.js project:
+    ```javascript
+    import CommonConfig from "eslint-config-yscope/CommonConfig.mjs";
+    import ReactConfigArray from "eslint-config-yscope/ReactConfigArray.mjs";
+    import StylisticConfigArray from "eslint-config-yscope/StylisticConfigArray.mjs";
+    import TsConfigArray from "eslint-config-yscope/TsConfigArray.mjs";
+    
+    
+    const EslintConfig = [
+        CommonConfig,
+    
+        // Note below map has to be placed directly before the spreaded `StylisticConfigArray`
```

Comment:
Why must it be placed before?

Reply:
Supposedly, if we manually configure individual rules in `TsConfigArray`, that would not be an issue. However, the current setup extends from a "recommended" preset `TsEslint.configs.strictTypeChecked`, which contains rules that might conflict with Stylistic rules.

The `StylisticConfigArray` contains a preset `StylisticPlugin.configs["disable-legacy"]` which contains rules to override any of those configured "recommended" rules as "off". By placing the `StylisticConfigArray` after `TsConfigArray`, the overrides will be correctly performed.

Reply:
Can you add that to the comment?